src/irmin-pack/inode.ml

```diff
@@ -302,7 +306,24 @@ struct
       in [irmin/slice.ml] and [irmin/sync_ext.ml], and maybe some other places.
 
       At some point we might want to forbid such deserialisations and instead
-      use something in the flavour of [Val.of_bin] to create [Partial] inodes. *)
+      use something in the flavour of [Val.of_bin] to create [Partial] inodes.
+
+      {3 Topmost Inode Ancestor}
+
+      [Val_impl.t] is a recursive type. The [depth] interer is a witness of the
+      recursion depth. An inode with [depth = 0] corresponds to the root of a
+      directory, its hash is the hash of the directory.
+
+      A [Val.t] pointes to the topmost [Val_impl.t] of an inode tree. In most
+      scenarios, that topmost inode has [depth = 0], but it is also legal for
+      the topmost inode to be an intermediate inode, i.e. with [depth > 0].
+
+      The only way for an inode tree to have an intermediate inode as root, is
+      to fetch it from the backend by calling [Make_ext.find], using the hash
+      that inode.
+
+      Write-only operations on trees starting by an intermediate inode are
```

Comment:
```suggestion
      Write-only operations on non-root inodes are
```

Reply:
I don't like the `root inode` term because it is ambiguous. Unless when the context makes it clear, it may refer to the root of the directory or the topmost ancestor of an inode tree.